app/src/pages/inside/stepPage/modals/editDefectModals/makeDecisionModal/optionsStepForm/optionsStepForm.jsx

```diff
@@ -14,86 +14,124 @@
  * limitations under the License.
  */
 
-import React, { useState } from 'react';
+import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { useSelector } from 'react-redux';
-import classNames from 'classnames/bind';
 import { useIntl } from 'react-intl';
-import { defectTypesSelector } from 'controllers/project';
 import { Accordion, useAccordionTabsState } from 'pages/inside/common/accordion';
-import {
-  CURRENT_EXECUTION_ONLY,
-  SOURCE_DETAILS,
-} from 'pages/inside/stepPage/modals/editDefectModals/constants';
-import { InputRadioGroup } from 'components/inputs/inputRadioGroup';
-import { LogItem } from 'pages/inside/logsPage/defectEditor/logItem';
-import { SourceDetails } from './sourceDetails';
+import { URLS } from 'common/urls';
+import { ERROR } from 'common/constants/logLevels';
+import { fetch } from 'common/utils';
+import { NOTIFICATION_TYPES, showNotification } from 'controllers/notification';
+import { useDispatch, useSelector } from 'react-redux';
+import { activeProjectSelector } from 'controllers/user';
+import { activeFilterSelector } from 'controllers/filter';
 import { messages } from './../../messages';
-import styles from './optionsStepForm.scss';
-
-const cx = classNames.bind(styles);
+import { OptionsSection } from './optionsSection';
+import { SourceDetails } from './sourceDetails';
+import { SEARCH_MODES, SOURCE_DETAILS } from '../../constants';
 
-export const OptionsStepForm = ({ info, itemData }) => {
+export const OptionsStepForm = ({ currentTestItem, modalState, setModalState }) => {
   const { formatMessage } = useIntl();
-  const defectTypes = Object.values(useSelector(defectTypesSelector)).flat();
-
-  const [tab, toggleTab] = useAccordionTabsState({
+  const dispatch = useDispatch();
+  const activeProject = useSelector(activeProjectSelector);
+  const activeFilter = useSelector(activeFilterSelector);
+  const [tab, toggleTab, collapseTabsExceptCurr] = useAccordionTabsState({
     [SOURCE_DETAILS]: true,
   });
-  const [optionValue, setOptionValue] = useState(CURRENT_EXECUTION_ONLY);
-  const options = [
-    {
-      ownValue: CURRENT_EXECUTION_ONLY,
-      label: {
-        id: CURRENT_EXECUTION_ONLY,
-        defaultMessage: formatMessage(messages.currentExecutionOnly),
-      },
-    },
-  ];
+  const { source, optionValue, loading, testItems, selectedItems } = modalState;
+  useEffect(() => {
+    const fetchLogs = (searchMode) => {
+      const requestData = {
+        searchMode,
+      };
+      if (searchMode === SEARCH_MODES.WITH_FILTER) {
+        requestData.filterId = activeFilter.id;
+      }
+      setModalState({
+        loading: true,
```

Comment:
Maybe it will be a local state? What's the purpose of keeping it on modal level if it's not used there.